sdk/core/Azure.Core/src/Request.cs

```diff
@@ -27,46 +25,53 @@ public new virtual RequestUriBuilder Uri
         {
             get
             {
-                return _uri ??= new RequestUriBuilder();
+                return _uriBuilder ??= new RequestUriBuilder();
             }
             set
             {
                 Argument.AssertNotNull(value, nameof(value));
-                _uri = value;
+                _uriBuilder = value;
             }
         }
 
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        /// <returns></returns>
+        protected override Uri GetUri() => Uri.ToUri();
```

Comment:
This whole business is kind of yucky, but will probably require larger generator changes to rework, and is a bit larger.  It's the second biggest thing in this PR that I'd like to rework